lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
-  this.allowedTypeOptions = ['shapeTypes'];
-}
-
-util.inherits(GeoShapeType, BaseConstructor);
-
-/**
- * @param {TypeOptions} typeOptions
- * @param {GeoShape} fieldValue
- * @param {string[]} errorMessages
- */
-GeoShapeType.prototype.validate = function geoShapeValidate (typeOptions, fieldValue, errorMessages) {
-  return this.recursiveShapeValidation(typeOptions.shapeTypes, fieldValue, errorMessages);
-};
+class GeoShapeType extends BaseType {
+  constructor() {
+    super();
+    this.typeName = 'geo_shape';
+    this.allowChildren = false;
+    this.allowedTypeOptions = ['shapeTypes'];
+  }
 
-/**
- * @param {string[]} allowedShapes
- * @param {GeoShape} shape
- * @param {string[]} errorMessages
- * @returns {boolean}
- */
-GeoShapeType.prototype.recursiveShapeValidation = function geoShapeRecusriveShapeValidation (allowedShapes, shape, errorMessages) {
-  var
-    isMulti = mutliTypes.indexOf(shape.type) !== -1,
-    coordinateValidation,
-    result = true,
-    i;
-
-  if (!this.checkStructure(allowedShapes, shape, errorMessages)) {
-    return false;
+  /**
+   * @param {TypeOptions} typeOptions
+   * @param {GeoShape} fieldValue
+   * @param {string[]} errorMessages
+   */
+  validate(typeOptions, fieldValue, errorMessages) {
+    return this.recursiveShapeValidation(typeOptions.shapeTypes, fieldValue, errorMessages);
   }
-  
-  switch (shape.type) {
-    case 'point':
-    case 'multipoint':
-      coordinateValidation = isPoint;
-      break;
-    case 'linestring':
-    case 'multilinestring':
-      coordinateValidation = isLine;
-      break;
-    case 'polygon':
-    case 'multipolygon':
-      coordinateValidation = isPolygon;
-      if (shape.orientation && allowedOrientations.indexOf(shape.orientation) === -1) {
-        errorMessages.push('The orientation property has not a valid value.');
-        result = false;
-      }
-      break;
-    case 'geometrycollection':
-      coordinateValidation = () => true;
 
-      for (i = 0; i < shape.geometries.length; i++) {
-        if (!this.recursiveShapeValidation(allowedShapes, shape.geometries[i], errorMessages)) {
+  /**
+   * @param {string[]} allowedShapes
+   * @param {GeoShape} shape
+   * @param {string[]} errorMessages
+   * @returns {boolean}
+   */
+  recursiveShapeValidation(allowedShapes, shape, errorMessages) {
+    const isMulti = multiTypes.indexOf(shape.type) !== -1;
+    let
+      coordinateValidation,
+      result = true;
+
+    if (!this.checkStructure(allowedShapes, shape, errorMessages)) {
+      return false;
+    }
+
+    switch (shape.type) {
+      case 'point':
+      case 'multipoint':
+        coordinateValidation = isPoint;
+        break;
+      case 'linestring':
+      case 'multilinestring':
+        coordinateValidation = isLine;
+        break;
+      case 'polygon':
+      case 'multipolygon':
+        coordinateValidation = isPolygon;
+        if (shape.orientation && allowedOrientations.indexOf(shape.orientation) === -1) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 89 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)